### ISSUE TITLE
ARIA Accessibility Compliance

### ISSUE THREAD

**Issue by UsernameIsInUse** [jfbrennan/m-#157]
I haven't had time to go through all of the custom design patterns for accessibility concerns, but the `m-alert` should have `role="alert"` to inform screen readers to stop and read the alert. Obviously going `<m-alert role="alert">` feels kind of silly, and the solution I can think of would be to set the role to alert when creating and registering the element class, but since this is CSS-first it's not built-in. Was curious about your thoughts to solve this accessibility issue.

**Comment by jfbrennan** [comment]
yeah the redundancy does feel goofy at first, but an alert role is actually not good for all types of alerts. Only alerts that are truly urgent or time-sensitive for users should have this role, this is noted in the component's a11y guidelines (see https://www.mdash.dev/#alert-a11y, also https://w3c.github.io/aria/#alert for full details).

Mdash, which follows the TAC CSS methodology, approaches ARIA like this:
- Require it as part of the API where absolutely required, which is rare when leveraging semantic HTML as Mdash does. A real example is the Switch component which requires `role="switch"`.
- Point developers in the right direction. An example of this is the Alert component, which documents and encourages basic a11y for alerts.
- Get out of the way, i.e. don't do anything that prevents a developer from applying ARIA to Mdash components
